### PR TITLE
Gallery UI tweaks

### LINKED ISSUE #4742: Gallery: hide calculator from gallery
Calculator is cool, but we don't have the cycles to make this super beautiful in time for the gallery's first release.

Let's hide it for now.

### PR DESCRIPTION
**Pull request by HansMuller**
Gallery home page cosmetic changes:
- Replace the home page's two-level list with a single list with category subheaders.
- Added a subtitle to each demo.
- Pad the Shrine order quantity dropdown
- Hide the calculator demo for now

Fixes #4742

### CODE REVIEW

File: examples/flutter_gallery/lib/gallery/home.dart [hunk 1]
```diff
@@ -75,26 +103,9 @@ class GalleryHomeState extends State<GalleryHome> {
...
-            title: new Text('Components'),
-            children: _componentItems
-          ),
-          new TwoLevelSublist(
-            leading: new Icon(Icons.color_lens),
-            title: new Text('Style'),
-            children: _styleItems
-          )
-        ]
+      body: new Block(
```
Comment (abarth) [inline examples/flutter_gallery/lib/gallery/home.dart]: LazyBlock ?

File: examples/flutter_gallery/lib/gallery/home.dart [hunk 2]
```diff
@@ -48,10 +48,38 @@ class GalleryHome extends StatefulWidget {
...
+        category = galleryItem.category;
+      }
+      _listItems.add(galleryItem);
+    }
+  }
 
   @override
   Widget build(BuildContext context) {
-    final double statusBarHight = (MediaQuery.of(context)?.padding ?? EdgeInsets.zero).top;
+    final double _statusBarHight = (MediaQuery.of(context)?.padding ?? EdgeInsets.zero).top;
```
Comment (abarth) [inline examples/flutter_gallery/lib/gallery/home.dart]: Why use an `_` for a local variable?

Also "Height" rather than "Hight" probably.

**Comment by abarth** [comment]
LGTM

**Comment by HansMuller** [comment]
I wasn't able to substitute LazyBlock for block directly because LazyBlock doesn't handle padding the same way that Block does.  Using a "spacer" list item zero, instead of padding the top of the list did work.

**Comment by abarth** [comment]
I think I've turned off the coveralls spam properly now.

**Comment by Hixie** [comment]
Please file a bug on the Block/LazyBlock difference, they should be drop-in replacements for each other (modulo the child list vs delegate thing).

**Comment by HansMuller** [comment]
I've filed two issues about LazyBlock https://github.com/flutter/flutter/issues/4880 https://github.com/flutter/flutter/issues/4879

**Pull request merged**